utils/load-NSF-profiles.py

```diff
@@ -0,0 +1,148 @@
+
+    try:
+        client.post_group(tilde_group)
+    except openreview.OpenReviewException as e:
+        print "Failed to post {0} group: {1}".format(tilde_id, e)
+        return
+
+    email_group = openreview.Group(email,
+                                   signatures=["OpenReview.net"],
+                                   signatories=[email],
+                                   readers=[email],
+                                   writers=[email],
+                                   nonreaders=[],
+                                   members=[tilde_id])
+    client.post_group(email_group)
+
+    profile = openreview.Note.from_json({
+        'id': tilde_id,
+        'content': {
+            'emails': [email],
+            'preferred_email': email,
+            'names': [{
+                'first': first,
+                'middle': '',
+                'last': last,
+                'username': tilde_id
+            }]
+        }
+    })
+
+    response = requests.put(client.baseurl + "/user/profile", json=profile.to_json(), headers=client.headers)
+
+# load information for one or more investigators from each xml file
+def load_xml_investigators(dirpath):
+    # get list of files in directory
+    file_names = os.listdir(dirpath)
+    investigators = {}
+    for filename in file_names:
+        if filename.endswith('.xml'):
+            f = open(dirpath+filename, 'r')
+            contents = ET.parse(f)
+            root = contents.getroot()
+            # for each <Investigator>
+            for person in root.iter('Investigator'):
+                email = person.find('EmailAddress').text
+                if email != None:
+                    email = email.strip()
+                    email = email.lower()
+                    if email not in investigators:
+                        # new investigator includes first, last name
+                        investigators[email]={}
+                        investigators[email]['first'] = person.find('FirstName').text
+                        investigators[email]['last'] = person.find('LastName').text
+                        investigators[email]['institute'] = ''
+                    if person.find('RoleCode').text == 'Principal Investigator':
+                        # we can only be sure the Institution is correct for PIs
+                        for institute in root.iter('Institution'):
+                            investigators[email]['institute'] = institute.find('Name').text
+            f.closed
+    return investigators
+
+# for each new name/email, create a new profile
+def upload_profiles(client, investigators):
+    for email in investigators:
+        # try to find profile w/ email
+        profile = get_profile(client, email)
+        # if this email already has a profile, no need to do anything
+        # if not, if find profile w/~name
+        if profile == None:
+            tilde_id = "~{0}_{1}1".format(investigators[email]['first'], investigators[email]['last'])
```

Comment:
the create_profile that is in the tools module creates the tilde id so you don't need to do it here.

Reply:
That eliminates half the code!
Is there a way to add institutions to profiles?

Reply:
Not now. I think we need to add a new revision to the profile with the new institution.  But this is not implemented yet.